robotiq_driver/CMakeLists.txt

```diff
@@ -23,88 +26,115 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
   serial
 )
 
+# Robotiq driver library.
+
 add_library(
-  ${PROJECT_NAME}
+  robotiq_driver
   SHARED
-  src/crc.cpp
+  include/robotiq_driver/fake/fake_driver.hpp
+  include/robotiq_driver/crc_utils.hpp
+  include/robotiq_driver/data_utils.hpp
+  include/robotiq_driver/default_driver.hpp
+  include/robotiq_driver/default_driver_factory.hpp
+  include/robotiq_driver/default_serial.hpp
+  include/robotiq_driver/default_serial_factory.hpp
+  include/robotiq_driver/driver.hpp
+  include/robotiq_driver/driver_exception.hpp
+  include/robotiq_driver/driver_factory.hpp
+  include/robotiq_driver/hardware_interface.hpp
+  include/robotiq_driver/serial.hpp
+  include/robotiq_driver/serial_factory.hpp
+  include/robotiq_driver/visibility_control.hpp
+  src/crc_utils.cpp
+  src/data_utils.cpp
   src/hardware_interface.cpp
-  src/robotiq_gripper_interface.cpp
+  src/default_driver.cpp
+  src/default_driver_factory.cpp
+  src/fake/fake_driver.cpp
+  src/default_serial.cpp
+  src/default_serial_factory.cpp
```

Comment:
I haven't changed the logic of how the driver works, I broke it apart into smaller pieces to allow testing without being connected to the hardware. Originally it was a monolithic project with no tests and with the hard requirement of having hardware at hand.

I intentionally followed the structure of the Epick Gripper so that later we can merge the two drivers into the same repo and extract common code. For example, all the serial communication is common to both.

I definitely made the mistake of formatting the code later which make the PR hard to read.

Reply:
Sorry this was just a note to self when I was reviewing I wanted to check the main changes parts in more detain.

I definitely agree we should get he Epick Grippper merged into one repo that would be great.